tests/test_coverity.py

```diff
@@ -1,126 +1,207 @@
-
-        # Test CoverityDefectService
-        coverity_service = cov.CoverityDefectService(coverity_conf_service)
-        suds_client_mock.assert_called_with('http://scan.coverity.com:8080/ws/v9/defectservice?wsdl')
-
-        coverity_service.login('user', 'password')
-        suds_security_mock.assert_called_once()
-        suds_username_mock.assert_called_once_with('user', 'password')
-
-        filters = {
-            'checker': None,
-            'impact': None,
-            'kind': None,
-            'classification': None,
-            'action': None,
-            'component': None,
-            'cwe': None,
-            'cid': None,
+        """SetUp to be run before each test to provide clean working env"""
+
+    def initialize_coverity_service(self, login=False):
+        """Logs in Coverity Service and initializes the urls used for REST API.
+
+        Returns:
+            CoverityDefectService: The coveritye defect service
+        """
+        coverity_conf_service = mlx.coverity.CoverityDefectService("scan.coverity.com/")
+
+        if login:
+            # Login to Coverity
+            coverity_conf_service.login("user", "password")
+
+        # urls that are used in GET or POST requests
+        endpoint = coverity_conf_service.api_endpoint
+        params = {
+            "queryType": "bySnapshot",
+            "retrieveGroupByColumns": "false"
+        }
+        self.column_keys_url = f"{endpoint}/issues/columns?{urlencode(params)}"
+        self.checkers_url = f"{endpoint}/checkerAttributes/checker"
+        self.stream_url = f"{endpoint}/streams/{self.fake_stream}"
+        params = {
+            "includeColumnLabels": "true",
+            "offset": 0,
+            "queryType": "bySnapshot",
+            "rowCount": -1,
+            "sortOrder": "asc",
+        }
+        self.issues_url = f"{endpoint}/issues/search?{urlencode(params)}"
+
+        return coverity_conf_service
+
+    @patch("mlx.coverity_services.requests")
+    def test_session_login(self, mock_requests):
+        """Test login function of CoverityDefectService"""
+        mock_requests.return_value = MagicMock(spec=requests)
+
+        # Get the base url
+        coverity_conf_service = mlx.coverity_services.CoverityDefectService("scan.coverity.com/")
+        self.assertEqual("https://scan.coverity.com/api/v2", coverity_conf_service.api_endpoint)
+
+        # Login to Coverity
+        coverity_conf_service.login("user", "password")
+        mock_requests.Session.assert_called_once()
+
+    @patch.object(mlx.coverity_services.requests.Session, "get")
+    def test_retrieve_checkers(self, mock_get):
+        """Test retrieving checkers (CoverityDefectService)"""
+        coverity_conf_service = mlx.coverity_services.CoverityDefectService("scan.coverity.com/")
+
+        # Login to Coverity
+        coverity_conf_service.login("user", "password")
+
+        with open("tests/columns_keys.json", "rb") as content:
```

Comment:
Please use `pathlib` to support Windows and make sure that this test can be executed from anywhere instead of only in the root of the repo.

Reply:
Resolved in 2abc4bfb624c1ea8298fe7013be6e8a03dfc675c

Reply:
> Changed in version 3.6: Support added to accept objects implementing [os.PathLike](https://docs.python.org/3/library/os.html#os.PathLike).

https://docs.python.org/3/library/functions.html#open

`open(TEST_FOLDER / columns_keys.json, "r", encoding="utf-8")` and I tend to use `TEST_DIR` as a variable name.